crates/inspector/src/inspect.rs

```diff
@@ -76,3 +77,88 @@ pub trait InspectCommitEvm: InspectEvm + ExecuteCommitEvm {
         Ok(output)
     }
 }
+
+/// InspectSystemCallEvm is an API that allows inspecting system calls in the EVM.
+///
+/// It extends [`InspectEvm`] and [`SystemCallEvm`] traits to provide inspection
+/// capabilities for system transactions, enabling tracing and debugging of
+/// system calls similar to regular transactions.
+pub trait InspectSystemCallEvm: InspectEvm + SystemCallEvm {
+    /// Inspect a system call with the current inspector.
+    ///
+    /// Similar to [`InspectEvm::inspect_one_tx`] but for system calls.
+    /// Uses [`SYSTEM_ADDRESS`] as the caller.
+    fn inspect_system_call_one(
```

Comment:
Inspection naming is `inspect_one_tx` so it would be nicer for this to be `inspect_one_system_call`